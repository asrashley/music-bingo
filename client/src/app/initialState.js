import { initialState as cards } from '../cards/cardsSlice';
import { initialState as games } from '../games/gamesSlice';
import { initialState as tickets } from '../tickets/ticketsSlice';
import { initialState as user } from '../user/userSlice';

export const initialState = Object.freeze({
    cards,
    games,
    tickets,
    user,
});
