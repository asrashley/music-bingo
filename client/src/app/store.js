import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import { connectRouter } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { reducer as formReducer } from 'redux-form';

import cardsReducer from '../cards/cardsSlice';
import gamesReducer from '../games/gamesSlice';
import ticketsReducer from '../tickets/ticketsSlice';
import userReducer from '../user/userSlice';
import adminReducer from '../admin/adminSlice';
import usersMiddleware from '../user/userMiddleware';

export const history = createBrowserHistory();

export const store = configureStore({
  middleware: [...getDefaultMiddleware(), usersMiddleware, logger],
  reducer: {
    admin: adminReducer,
    cards: cardsReducer,
    form: formReducer,
    games: gamesReducer,
    tickets: ticketsReducer,
    router: connectRouter(history),
    user: userReducer,
  },
});
