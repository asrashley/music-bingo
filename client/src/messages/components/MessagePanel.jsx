import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Message } from './Message';
import { clearMessage } from '../messagesSlice';
import { getMessages } from '../messagesSelectors';

import '../styles/messages.scss';

class MessagePanel extends React.Component {
  static propTypes = {
    messages: PropTypes.array.isRequired,
    dispatch: PropTypes.func.isRequired,
  };

  clearMessage = (msg) => {
    const { dispatch } = this.props;
    dispatch(clearMessage(msg));
  }

  render() {
    const { messages } = this.props;
    return (
      <div id="message-panel" >
        {messages.map((msg, key) => <Message key={key} msg={msg} clearMessage={this.clearMessage}/>)}
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    messages: getMessages(state, props),
  };
};

MessagePanel = connect(mapStateToProps)(MessagePanel);

export {
  MessagePanel
};
