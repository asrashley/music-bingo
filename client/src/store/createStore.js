import { configureStore } from '@reduxjs/toolkit';
import { createLogger } from 'redux-logger';
import { createRouterMiddleware, createRouterReducer } from '@lagunovsky/redux-react-router';
import log from 'loglevel';

import adminReducer from '../admin/adminSlice';
import directoriesReducer from '../directories/directoriesSlice';
import gamesReducer from '../games/gamesSlice';
import messagesReducer from '../messages/messagesSlice';
import routesReducer from '../routes/routesSlice';
import settingsReducer from '../settings/settingsSlice';
import systemReducer from '../system/systemSlice';
import ticketsReducer from '../tickets/ticketsSlice';
import userReducer from '../user/userSlice';
import usersMiddleware from '../user/userMiddleware';
import messagesMiddleware from '../messages/messagesMiddleware';

import { history } from './history';
const routerMiddleware = createRouterMiddleware(history);

export function createStore(preloadedState, logging) {
  const middleware = (getDefaultMiddleware) => {
    const mw = getDefaultMiddleware()
      .prepend(routerMiddleware)
      .concat(usersMiddleware)
      .concat(messagesMiddleware);
    if (logging === undefined) {
      logging = import.meta.env.DEV;
    }
    if (!logging) {
      return mw;
    }
    const logger = {
      debug: log.debug,
      info: log.info,
      warn: log.warn,
      error: log.error,
      log: log.debug,
    };
    const loggerMiddleware = createLogger({
      level: 'debug',
      logger
    });
    return mw.concat(loggerMiddleware);
  };
  return configureStore({
    middleware,
    reducer: {
      admin: adminReducer,
      directories: directoriesReducer,
      games: gamesReducer,
      messages: messagesReducer,
      routes: routesReducer,
      settings: settingsReducer,
      system: systemReducer,
      tickets: ticketsReducer,
      router: createRouterReducer(history),
      user: userReducer,
    },
    preloadedState
  });
}
