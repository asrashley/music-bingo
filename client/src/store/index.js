export { createStore } from './createStore';
export { initialState } from './initialState';
export { history } from './history';
