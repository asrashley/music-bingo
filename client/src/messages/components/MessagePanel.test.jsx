import React from 'react';

import { renderWithProviders } from '../../../tests';
import { MessagePanel, MessagePanelComponent } from './MessagePanel';

import { initialState } from '../../store/initialState';

describe('MessagePanel component', () => {
	const messages = [{
		id: 2,
		text: "a single line message",
		heading: "message heading",
		hidden: false,
		type: "success",
		timestamp: 1675888498892
	}, {
		id: 3,
		text: "message 2",
		heading: "message heading 2",
		hidden: false,
		type: "info",
		timestamp: 1675888498900
	}];

	beforeAll(() => {
		vi.useFakeTimers('modern');
		vi.setSystemTime(new Date('08 Feb 2023 20:34:58 GMT').getTime());
	});

	afterAll(() => vi.useRealTimers());

	it('MessagePanelComponent renders a list of messages', () => {
		const result = renderWithProviders(
			<MessagePanelComponent messages={messages} dispatch={() => false} />);
		messages.forEach((msg) => {
			result.getByText(msg.text);
			result.getByText(msg.heading);
		});
	});

	it('fetches messages from the Redux store', () => {
		const preloadedState = {
			...initialState,
			messages: {
				messages: {},
				duration: 15000,
				nextMessageId: 4,
			}
		};
		messages.forEach(msg => preloadedState.messages.messages[msg.id] = msg);
		const result = renderWithProviders(
			<MessagePanel />, { preloadedState });
		messages.forEach((msg) => {
			result.getByText(msg.text);
			result.getByText(msg.heading);
		});
	});

	it('matches JSON snapshot', () => {
		const { asFragment } = renderWithProviders(
			<MessagePanelComponent messages={messages} dispatch={() => false} />);
		expect(asFragment()).toMatchSnapshot();
	});
});