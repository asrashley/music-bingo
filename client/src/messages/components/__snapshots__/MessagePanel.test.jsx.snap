// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MessagePanel component > matches JSON snapshot 1`] = `
<DocumentFragment>
  <div
    id="message-panel"
  >
    <div
      class="alert alert-dismissible alert-success"
      id="msg_2"
      role="alert"
    >
      <button
        aria-label="Close"
        class="close"
        data-dismiss="alert"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
      <div
        class="message-timestamp"
      >
        20:34:58 08/02/2023
      </div>
      <span
        aria-hidden="true"
        class="glyphicon glyphicon-ok-sign"
      >
         
      </span>
      <div
        class="message-text"
      >
        <h4
          class="alert-heading"
        >
          message heading
        </h4>
        <p>
          a single line message
        </p>
      </div>
    </div>
    <div
      class="alert alert-dismissible alert-info"
      id="msg_3"
      role="alert"
    >
      <button
        aria-label="Close"
        class="close"
        data-dismiss="alert"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
      <div
        class="message-timestamp"
      >
        20:34:58 08/02/2023
      </div>
      <span
        aria-hidden="true"
        class="glyphicon glyphicon-info-sign"
      >
         
      </span>
      <div
        class="message-text"
      >
        <h4
          class="alert-heading"
        >
          message heading 2
        </h4>
        <p>
          message 2
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MessagePanel component matches JSON snapshot 1`] = `
<DocumentFragment>
  <div
    id="message-panel"
  >
    <div
      class="alert alert-dismissible alert-success"
      id="msg_2"
      role="alert"
    >
      <button
        aria-label="Close"
        class="close"
        data-dismiss="alert"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
      <div
        class="message-timestamp"
      >
        20:34:58 08/02/2023
      </div>
      <span
        aria-hidden="true"
        class="glyphicon glyphicon-ok-sign"
      >
         
      </span>
      <div
        class="message-text"
      >
        <h4
          class="alert-heading"
        >
          message heading
        </h4>
        <p>
          a single line message
        </p>
      </div>
    </div>
    <div
      class="alert alert-dismissible alert-info"
      id="msg_3"
      role="alert"
    >
      <button
        aria-label="Close"
        class="close"
        data-dismiss="alert"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
      <div
        class="message-timestamp"
      >
        20:34:58 08/02/2023
      </div>
      <span
        aria-hidden="true"
        class="glyphicon glyphicon-info-sign"
      >
         
      </span>
      <div
        class="message-text"
      >
        <h4
          class="alert-heading"
        >
          message heading 2
        </h4>
        <p>
          message 2
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
